### DISCUSSION TITLE
Recommeded way to add new environments/tasks/agents etc.?

### DISCUSSION THREAD

**Discussion by meghbhalerao** [isaac-sim/IsaacLab#1354] [Q&A] [answered]
I am trying to add some custom assets and custom multi agent tasks, and I (think, from what I understand), I have the following options to do this - 
1. I use isaaclab as any other python package and import necessary envs etc, and maintain a seperate dir structure for my custom assets and tasks, and if necessary do inheritances from the original python package.
2. I have a local copy of isaaclab which is cloned from the main repo, or maybe my personal fork, and I make modifications to that, i.e add new assets and envs and tasks to the folders within isaac lab for example here 
- https://github.com/isaac-sim/IsaacLab/tree/main/source/extensions/omni.isaac.lab_assets/omni/isaac/lab_assets for assets 
- https://github.com/isaac-sim/IsaacLab/tree/main/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks for tasks, either manager or direct based tasks.
3. I use the extension template here https://github.com/isaac-sim/IsaacLabExtensionTemplate (I didn't quite understand this, is this for adding new tasks, envs, assets etc?)
It would be great if I can get some suggestions on this, thanks for your time!

**Comment by RandomOakForest** [comment, marked as answer]
Great question. Either of the 3 options will work. The extension template (3) is a way of isolating user code from Isaac Lab, similar to (1), but provides a template that can be run as an Omniverse extension.
